pkg/testcase/secretsencrypt.go

```diff
@@ -0,0 +1,170 @@
+package testcase
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/rancher/distros-test-framework/shared"
+
+	. "github.com/onsi/gomega"
+)
+
+func TestSecretsEncryption() {
+	nodes, errGetNodes := shared.GetNodesByRoles("etcd", "control-plane")
+	Expect(nodes).NotTo(BeEmpty())
+	Expect(errGetNodes).NotTo(HaveOccurred(), "error getting etcd/control-plane nodes")
+
+	product, err := shared.Product()
+	Expect(err).NotTo(HaveOccurred(), "error getting product from config")
+
+	ips := getNodeIps(nodes)
```

Comment:
why getnodeIps if your  `nodes` already has it? 

```
	nodes, errGetNodes := shared.GetNodesByRoles("etcd", "control-plane")

	for _, ip := range nodes {
		ip.ExternalIP = ""

	}

```

Reply:
just making my life easier, working with a simple string array, instead of using the whole object. Every time, i ll hv to reference, and look for the external ip. also, the shared methods take string arrays as inputs. just thought, keep the datatype needed simple and uniform across all methods.

Reply:
sorry but this is creating extra functions unnecessary you already have the data u need....on pov makes no sense cause u are already also using it bellow
```
	secretsEncryptOps("prepare", product, nodes[index].ExternalIP, ips)
```

Reply:
but thats up to you

Reply:
so the original getips method, was because, i was getting the etcd nodes first and then the control plane nodes, and getting their individual ips and stitching them together. but after Max suggested i can get them together, this method became simple. But all methods here and the shared methods are using the ips array. If I modify it now, it will be a major rework of all the methods/calls and retesting. I wont be able to complete it this cycle. Will have to get pushed later. I would like to leave this as is for now.

Reply:
so the original getips method, was because, i was getting the etcd nodes first and then the control plane nodes, and getting their individual ips and stitching them together. but after Max suggested i can get them together, this method became simple. But all methods here and the shared methods are using the ips array. If I modify it now, it will be a major rework of all the methods/calls and retesting. I wont be able to complete it this cycle. Will have to get pushed later. I would like to leave this as is for now.